selfie.c

```diff
@@ -4104,6 +4104,171 @@ int* copyC2CStarArguments(int argc, int *argv) {
     return cstar_argv;
 }
 
+/////////////////////////////////////////////////////////////////////
+//////------Assignment 0: Implementation of a Linked List------//////
+/////////////////////////////////////////////////////////////////////
+
+// Initialization of the linked list (Creating the header with data -1)
+// The header cannot be deleted!!
+int* initializeList(int *list) {
+	int *header;
+	int data = -1; // no data inserted
+
+	header = malloc(4*2);
+	*header = 0;
+	*(header+1) = data;
+	return header;
+}
+
+// Add a node to the linked list at the bottom
+int* addToList(int *list, int data) {
+
+	int *newNode;
+	newNode = malloc(4*2);
+	*newNode = list;
+	*(newNode+1) = data;
+	return newNode;
+}
+
+// Iterate through the linked list and get the nth node of the list
+int* getNodeFromList(int* list, int nthNode) {
+
+	while(nthNode > 0) {
+		list = *list;
+		nthNode = nthNode - 1;
+	}
+
+	return list;
+}
+
+// Iterate through the linked list and get the data of the nth node of the list
+int getDataFromList(int* list, int nthNode) {
```

Comment:
Why duplicate the loop from getNodeFromList? One you've got the node, you got the data